dotCMS/src/integration-test/java/com/dotcms/contenttype/business/DotAssetBaseTypeToContentTypeStrategyImplTest.java

```diff
@@ -0,0 +1,118 @@
+package com.dotcms.contenttype.business;
+
+import com.dotcms.IntegrationTestBase;
+import com.dotcms.contenttype.model.field.BinaryField;
+import com.dotcms.contenttype.model.field.Field;
+import com.dotcms.contenttype.model.field.FieldVariable;
+import com.dotcms.contenttype.model.field.ImmutableFieldVariable;
+import com.dotcms.contenttype.model.type.BaseContentType;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.contenttype.model.type.ContentTypeBuilder;
+import com.dotcms.contenttype.model.type.DotAssetContentType;
+import com.dotcms.mock.request.MockAttributeRequest;
+import com.dotcms.mock.request.MockHeaderRequest;
+import com.dotcms.mock.request.MockHttpRequest;
+import com.dotcms.mock.request.MockSessionRequest;
+import com.dotcms.rest.api.v1.temp.DotTempFile;
+import com.dotcms.util.CollectionsUtils;
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.portlets.folders.business.FolderAPI;
+import com.dotmarketing.util.FileUtil;
+import com.dotmarketing.util.UUIDGenerator;
+import com.liferay.portal.model.User;
+import com.liferay.portal.util.PortalUtil;
+import com.liferay.portal.util.WebKeys;
+import org.glassfish.jersey.internal.util.Base64;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+
+public class DotAssetBaseTypeToContentTypeStrategyImplTest  extends IntegrationTestBase {
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * 1) creates a dotAsset content type for text plain
+     * 2) creates a temporal file asset and gets the id
+     * 3) creates a map with the contentlet properties (including as a file asset the temporal file id)
+     * 4) runs the BaseTypeToContentTypeStrategy in order to get the new content type using the temporal file
+     * @throws Exception
+     */
+    @Test
+    public void test_apply_with_temporal_file() throws Exception {
+
+        // creates a dotAsset for text files
+        final String variable = "testDotAsset" + System.currentTimeMillis();
+        final ContentType dotAssetContentType = APILocator.getContentTypeAPI(APILocator.systemUser()).
+                save(ContentTypeBuilder.builder(DotAssetContentType.class).folder(
+                FolderAPI.SYSTEM_FOLDER).host(Host.SYSTEM_HOST).name(variable)
+                .owner(APILocator.systemUser().getUserId()).build());
+
+        final Map<String, Field> fieldMap = dotAssetContentType.fieldMap();
+        com.dotcms.contenttype.model.field.Field binaryField           = fieldMap.get(DotAssetContentType.ASSET_FIELD_VAR);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'com.dotcms.contenttype.model.field.Field' due to existing import 'com.dotcms.contenttype.model.field.Field'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093431)